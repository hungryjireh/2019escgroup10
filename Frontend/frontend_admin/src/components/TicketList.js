import React from "react";
import TicketListItem from "./TicketListItem";
import "../style/TicketList.css";

const TicketList = ({ user, tickets }) => {
  const renderedTickets = tickets.map(ticket => {
    return <TicketListItem user={user} key={ticket.id} ticket={ticket} />;
  });

  return (
    <div className="ticketlist">
      <div className="ticketlist-header">
        <ul className="ticketlist-header-list">
          <li>Subject</li>
          <li>Status</li>
          <li>Requester</li>
          <li>Last Updated</li>
          <li>Group</li>
        </ul>
      </div>
      <div className="ticketlist-body">
        <ul className="ticketlist-body-list">{renderedTickets}</ul>
      </div>
    </div>
  );
};

export default {TicketList, renderedTickets};
