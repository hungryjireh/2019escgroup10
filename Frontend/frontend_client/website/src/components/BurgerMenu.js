import React from "react";
import styled from "styled-components";

import useLockBodyScroll from "./useLockBodyScroll";
import "./BurgerMenuAnimations.css";

const BurgerMenu = styled.div`
  width: 230px;
  height: 100vh;
  position: fixed;
  left: 0;
  padding: 2rem 3rem;
  display: flex;
  flex-direction: column;
  font-family: "Montserrat", sans-serif;
  background-color: white;
  z-index: 5;
`;

export const Darken = styled.div`
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: rgb(0, 0, 0);
  opacity: 0.4;
  z-index: 4;
  transition
`;

export const BurgerLink = styled.a`
  font-size: 1.4rem;
  margin-bottom: 3rem;
`;

const BurgerMenuComp = ({ setHasBurgerFalse }) => {
  useLockBodyScroll();
  return (
    <div>
      <BurgerMenu className="slide-in-animation">
        <BurgerLink href="https://beta.acnapi.io">Home</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#about_us">About Us</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#assets">Our Assets</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#case_study">Case Studies</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/">Contact Us</BurgerLink>
      </BurgerMenu>
      <Darken
        className="darken-animation"
        onClick={() => {
          document
            .querySelector(".slide-in-animation")
            .classList.toggle("slide-out-animation");
          document
            .querySelector(".darken-animation")
            .classList.toggle("lighten-animation");
          setTimeout(() => {
            setHasBurgerFalse();
          }, 400);
        }}
      />
    </div>
  );
};

export default BurgerMenuComp;
