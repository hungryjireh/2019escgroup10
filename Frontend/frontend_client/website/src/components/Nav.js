import React, { useState } from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";

import BurgerMenu from "./BurgerMenu";
import "./Nav.css";

const Nav = styled.nav`
  max-width: 1440px;
  font-size: 1.5rem;
  font-family: "Montserrat", sans-serif;
  background: #000000;
  position: fixed;
  width: 100%;
  z-index: 3;
`;

export const NavList = styled.ul`
  display: flex;
  align-items: center;
`;

export const ListItem = styled.li`
  padding: 1.5rem;
  margin-left: 3rem;
  cursor: pointer
    &:hover > a {
    color: #f9c03f;
  }
`;

export const ItemLink = styled.a`
  color: #fefefe;
  &::selection {
    background-color: #f9c03f;
    color: #2e2e2e;
  }
`;

export const NavBurger = styled.img`
  display: none;

  @media (max-width: 1100px) {
    display: inline;
    margin: 0 2.4rem 0 3rem;
    width: 20px;
    cursor: pointer;
  }
`;

const NavComp = () => {
  const [hasBurger, setHasBurger] = useState(false);

  const handleBurgerClick = () => {
    setHasBurger(true);
  };

  const setHasBurgerFalse = () => {
    setHasBurger(false);
  };

  return (
    <Nav>
      <NavList>
        <ListItem style={{ margin: "0 auto 0 1.6rem" }}>
          <ItemLink data-testid="home-logo" href="https://beta.acnapi.io">
            <img
              className="company-logo"
              src={require("../assets/acnapi-logo-white.png")}
              alt="company logo"
              style={{ width: "83px" }}
            />
          </ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#about_us">About Us</ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#assets">Our Assets</ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#case_study">Case Studies</ItemLink>
        </ListItem>
        <ListItem className="nav-links" style={{ margin: "0 1.6rem 0 3rem" }}>
          <ItemLink href="https://beta.acnapi.io/#!/">Contact Us</ItemLink>
        </ListItem>
        <NavBurger
          onClick={handleBurgerClick}
          src={require("../assets/burger-white.svg")}
          alt="nav burger"
        />
      </NavList>
      {hasBurger &&
        ReactDOM.createPortal(
          <BurgerMenu setHasBurgerFalse={setHasBurgerFalse} />,
          document.querySelector("#burger")
        )}
    </Nav>
  );
};

export default NavComp;
