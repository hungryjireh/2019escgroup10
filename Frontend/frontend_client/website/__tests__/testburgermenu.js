import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import BurgerMenu from "../components/BurgerMenu";
import {BurgerLink, Darken} from "../components/BurgerMenu";

afterEach(cleanup);

it("BurgerLink renders", () => {
    const { asFragment } = render(<BurgerLink />);
    expect(asFragment()).toMatchSnapshot();
});

test('Navbar elements all present', () => {
    const { getByText, getByAltText } = render(<BurgerMenu />);
    expect(getByText('Home'));
    expect(getByText('About Us'));
    expect(getByText('Case Studies'));
    expect(getByText('Our Assets'));
    expect(getByText('Contact Us'));
})

test('assert Home to be home link', () => {
    const { getByText } = render(<BurgerMenu />);
    expect(getByText('Home').getAttribute('href')).toContain('https://beta.acnapi.io');
})

test('assert About Us to be About Us link', () => {
    const { getByText } = render(<BurgerMenu />);
    expect(getByText('About Us').getAttribute('href')).toContain('https://beta.acnapi.io/#!/#about_us');
})

test('assert Our Assets to be Our Assets link', () => {
    const { getByText } = render(<BurgerMenu />);
    expect(getByText('Our Assets').getAttribute('href')).toContain('https://beta.acnapi.io/#!/#assets');
})

test('assert Case Studies to be Case Studies link', () => {
    const { getByText } = render(<BurgerMenu />);
    expect(getByText('Case Studies').getAttribute('href')).toContain('https://beta.acnapi.io/#!/#case_study');
})

test('calls "onClick" prop on button click of Darken', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Darken onClick={onClick} className="darken animation" data-testid="darken-animation"/>);

    fireEvent.click(getByTestId(/darken-animation/i));
    expect(onClick).toHaveBeenCalled();
});