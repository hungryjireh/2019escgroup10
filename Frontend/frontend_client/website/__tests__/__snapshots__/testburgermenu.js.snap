// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerLink renders 1`] = `
<DocumentFragment>
  <a
    class="sc-htpNat gOasEt"
  />
</DocumentFragment>
`;
